### ISSUE TITLE
Filter throws exception when some columns are hidden

### ISSUE THREAD

**Issue by coderazzi** [coderazzi/tablefilter-swing#83]
(Original issue 37 created by None on 2014-06-23T14:43:33+00:00)

```
What steps will reproduce the problem?
1. set up a table; hide some columns
2. set up filter header as normal
3. add a separate filter; the separate filter causes the crash

What is the expected output? What do you see instead?

filter the table; instead, it throws an exception.

What version of the product are you using? On what operating system?

5.1.1

Please provide any additional information below.

// create table and TableFilterHeader in the usual way
JTable table = ....
 TableFilterHeader filterHeader = new TableFilterHeader(_table, AutoChoices.ENABLED);
filterHeader.setFilterOnUpdates(true);

//now add an additional, separate filter to the header
filterHeader.addFilter(RowFilter.regexFilter(&quot;ALICE&quot;);

// the separate filter cause an exception b/c some columns are hidden (see Entry.getStringValue(...)

// if the visible columns are inserted into the regex filter, the exception is avoided
filterHeader.addFilter(RowFilter.regexFilter(&quot;ALICE&quot;,visibleColumnsIntArray);


```

Original issue reported on code.google.com by `elisebotworld.ww` on 2014-06-23 14:43:33

**Comment by coderazzi** [comment]
(Original issue 37 last updated on 2015-04-26T00:22:40.773211+00:00)

(Issue automaticaly closed due to status in Bitbucket: resolved)

**Comment by coderazzi** [comment]
(Original comment by {u'display_name': u'coderazzi coderazzi', u'account_id': u'557058:790eb198-2ae4-49b4-b920-8d654aa86b3f'} on 2014-06-25T11:26:56+00:00)

```
Hi, I was checking the code that you provided, but addFilter requires an IFilter instance,
not a RowFilter.
I still get an exception, as you point out, when I create an IFilter wrapping the RowFilter,
doing:

        filterHeader.addFilter(new Filter(){
    		RowFilter r = RowFilter.regexFilter("ALICE");
        	@Override public boolean include(Entry entry) {
        		return r.include(entry);
        	}
        });        		

Is this the code that you have?
```

Original issue reported on code.google.com by `coderazzi` on 2014-06-25 11:26:56

**Comment by coderazzi** [comment]
(Original comment by {u'display_name': u'coderazzi coderazzi', u'account_id': u'557058:790eb198-2ae4-49b4-b920-8d654aa86b3f'} on 2015-03-18T22:26:18+00:00)

```
Solved on 5.1.2
```

Original issue reported on code.google.com by `coderazzi` on 2015-03-18 22:26:18

**Comment by coderazzi** [comment]
(Original comment by {u'display_name': u'coderazzi coderazzi', u'account_id': u'557058:790eb198-2ae4-49b4-b920-8d654aa86b3f'} on 2015-04-26T00:22:40.782784+00:00)

Already resolved (previously, before migrating to BitBucket)
